playbooks/robusta_playbooks/job_restart_on_oomkilled.py

```diff
@@ -36,79 +57,308 @@ def job_restart_on_oomkilled(event: JobEvent, params: IncreaseResources):
+                    keep_the_same,
+                )
             )
-            event.add_finding(finding)
-    else:
-        finding.title = f" POD FAILED "
-        finding.add_enrichment(
-            [
-                MarkdownBlock(f"*The job's pod was not killed because of OOM*"),
-            ]
+
+    job_spec = job_fields(job_event, containers)
+    job_event.delete()
+    job_spec.create()
+
+    containers_memory_list = []
+
+    # Getting information for finding
+    for index, containers in enumerate(job_spec.spec.template.spec.containers):
+        containers_memory_list.append(containers.name)
+        containers_memory_list.append(containers.resources.requests["memory"])
+
+    finding.title = f" JOB RESTARTED"
+    finding.add_enrichment(
+        [
+            MarkdownBlock(
+                f"*containers memory after restart*\n```\n{containers_memory_list}\n```"
+            ),
+        ]
+    )
+    event.add_finding(finding)
+
+
+# Function to increase resource of the container
+def increase_resource(container, max_resource, increase_by, keep_the_same):
+    # Getting Container attributes
+    container = Container(
+        name=container.name,
+        image=container.image,
+        livenessProbe=container.livenessProbe,
+        securityContext=container.securityContext,
+        args=container.args,
+        command=container.command,
+        ports=container.ports,
+        lifecycle=container.lifecycle,
+        readinessProbe=container.readinessProbe,
+        workingDir=container.workingDir,
+        env=container.env,
+        startupProbe=container.startupProbe,
+        envFrom=container.envFrom,
+        imagePullPolicy=container.imagePullPolicy,
+        resources=memory_increment(
+            container.resources,
+            increase_by,
+            max_resource,
+            keep_the_same,
         )
-        event.add_finding(finding)
-
-# Function to restart job
-def restart_job(job_event, increase_by, max_resource):
-    container_list = get_container_list(
-        job_event.spec.template.spec.containers,
-        increase_by=increase_by,
-        max_resource=max_resource,
+        if (container.resources.limits and container.resources.requests)
+        else None,
     )
+    return container
+
+
+# Function to increment in memory
+def memory_increment(resources, increase_by, max_resource, keep_the_same):
+    if keep_the_same:
+        return resources
+    else:
+        limits = resources.limits["memory"]
+        reqests = resources.requests["memory"]
+
+        # splitting num and str
+        existing_limit_memory, existing_lim_unit = split_num_and_str(limits)
+        existing_req_memory, existing_req_unit = split_num_and_str(reqests)
+
+        split_increased_memory, split_increased_memory_unit = split_num_and_str(
+            increase_by
+        )
+
+        # Checking if provided unit is same as job's memory unit
+        if existing_req_unit == split_increased_memory_unit:
```

Comment:
I see you used bitmath, however the code repeats itself several times.
I saw bitmath have a "Rich Comparison" feature, allowing you to compare between Mib and Gib, for example.

I think the code can be much shorter if you try:
1. Getting the limit/request string values from the container.
2. Use bitmath.parse_string to convert the string to bitmath.
4. Use bitmath.parse_string to convert max_resource & increase_by value to bitmath.
5. Compare between the two.
6. If applicable add the request/limit to the increase_by using bitmath
7. Create ResourceRequirements using str()